src/components/AdminPanelLayout/UserListItem.tsx

```diff
@@ -0,0 +1,49 @@
+import React, {FormEvent, useState} from 'react';
+import {Role, User} from '../../types/User';
+
+export interface ChangeUserRoleRequestSchema {
+  newRole: string;
+  userId: number;
+}
+
+interface Props {
+  user: User;
+  onRoleChange: (r: ChangeUserRoleRequestSchema) => void;
+}
+
+const roleOptions = [
+  {value: 'Guest' as Role, label: 'Guest'},
+  {value: 'Creator' as Role, label: 'Creator'},
+  {value: 'Admin' as Role, label: 'Admin'}
+];
```

Comment:
go to User.ts

Replace 

```
export type Role = 'Guest' | 'Creator' | 'Admin';
```

with:

```
export const ERole = {
  'Guest': 'Guest',
  'Creator': 'Creator',
  'Admin': 'Admin'
} as const;
export type Role = keyof typeof ERole;
```

then make role Options dynamic so that we have less places we need to change to add a new role.

```suggestion
const roleOptions = Object.values(ERole).map(r => ({value: r, label: r}));
```